R/shinyui.R

```diff
@@ -150,8 +152,15 @@ uiHttpHandler <- function(ui, uiPattern = "^/$") {
     if (is.null(uiValue))
       return(NULL)
 
-    renderPage(uiValue, textConn, showcaseMode, testMode)
-    html <- paste(readLines(textConn, encoding = 'UTF-8'), collapse='\n')
-    return(httpResponse(200, content=enc2utf8(html)))
+    if (inherits(uiValue, "httpResponse")) {
+      return(uiValue)
+    } else {
+      textConn <- file(open = "w+")
```

Comment:
Looking at this code, I realized how awkward and unnecessary all this `textConn` stuff is. This function calls `renderPage()`, which calls `renderDocument()`. That function actually returns a string, but then then `renderPage` takes that string and writes it to a connection, which we read from here and then paste together into a string. We might as well take out the text connection entirely. But that can be another PR if you don't want to do it here.